x-pack/test/functional/page_objects/api_keys_page.ts

```diff
@@ -11,10 +11,19 @@ export function ApiKeysPageProvider({ getService }: FtrProviderContext) {
 
   return {
     async noAPIKeysHeading() {
-      return await testSubjects.find('noApiKeysHeader');
+      const messageElement = await testSubjects.find('noApiKeysHeader');
+      const message = await messageElement.getVisibleText();
+      return message;
     },
+
     async getGoToConsoleButton() {
       return await testSubjects.find('goToConsoleButton');
     },
+
+    async apiKeysPermissionDeniedMessage() {
+      const messageElement = await testSubjects.find('apiKeysPermissionDeniedMessage');
+      const message = await messageElement.getVisibleText();
```

Comment:
You can compress these 2 lines into one using 
`const message = await testSubjects.getVisibleText('apiKeysPermissionDeniedMessage')`

Reply:
incorporated the suggestions.